yapf/yapflib/style.py

```diff
@@ -375,18 +375,29 @@ def GlobalStyles():
     if not def_style:
       return _style
     return _GLOBAL_STYLE_FACTORY()
-  style_factory = _STYLE_NAME_TO_FACTORY.get(style_config.lower())
-  if style_factory is not None:
-    return style_factory()
-  if style_config.startswith('{'):
-    # Most likely a style specification from the command line.
-    config = _CreateConfigParserFromConfigString(style_config)
-  else:
-    # Unknown config name: assume it's a file name then.
-    config = _CreateConfigParserFromConfigFile(style_config)
+  if isinstance(style_config, dict):
```

Comment:
I think that would include counter and since all the values need to be strings for this to work properly, that seems dangerous? But maybe that means `isinstance` is the wrong way to go for this, since it allows subclasses to also count… maybe I should make a check explicitly that all the values are strings (or only pass through pairs to the config object if the value is a string)?